chia/full_node/full_node.py

```diff
@@ -2351,26 +2343,71 @@ async def add_transaction(
                 # vector.
                 mempool_item = self.mempool_manager.get_mempool_item(spend_name)
                 assert mempool_item is not None
-                fees = mempool_item.fee
-                assert fees >= 0
-                assert cost is not None
-                new_tx = full_node_protocol.NewTransaction(
-                    spend_name,
-                    cost,
-                    fees,
-                )
-                msg = make_msg(ProtocolMessageTypes.new_transaction, new_tx)
-                if peer is None:
-                    await self.server.send_to_all([msg], NodeType.FULL_NODE)
-                else:
-                    await self.server.send_to_all([msg], NodeType.FULL_NODE, peer.peer_node_id)
+
+                await self.notify_peers_of_mempool_item(mempool_item, peer)
+
                 if self.simulator_transaction_callback is not None:  # callback
                     await self.simulator_transaction_callback(spend_name)  # pylint: disable=E1102
+
             else:
                 self.mempool_manager.remove_seen(spend_name)
                 self.log.debug(f"Wasn't able to add transaction with id {spend_name}, status {status} error: {error}")
         return status, error
 
+    async def notify_peers_of_mempool_item(
+        self, mempool_item: MempoolItem, peer: Optional[WSChiaConnection] = None
+    ) -> None:
+        new_tx = full_node_protocol.NewTransaction(
+            mempool_item.spend_bundle_name,
+            mempool_item.cost,
+            mempool_item.fee,
+        )
+        new_tx_msg = make_msg(ProtocolMessageTypes.new_transaction, new_tx)
+
+        await self.server.send_to_all([new_tx_msg], NodeType.FULL_NODE, None if peer is None else peer.peer_node_id)
+
+        wallet_update = wallet_protocol.TransactionAddedUpdate(
+            mempool_item.spend_bundle_name,
+            mempool_item.height_added_to_mempool,
+            mempool_item.cost,
+            mempool_item.additions,
+            mempool_item.removals,
+        )
+        wallet_update_msg = make_msg(ProtocolMessageTypes.transaction_added_update, wallet_update)
+
+        peers = await self.peers_for_mempool_item(mempool_item)
```

Comment:
it would probably make sense to call this earlier and have an early exit in case `peers == []`

Reply:
Will be another PR